doc/source/terrain.md

```diff
@@ -10,8 +24,87 @@ and tested for consistency against [gdaldem](https://gdal.org/programs/gdaldem.h
 ## Quick use
 
 Terrain attribute methods can either be called directly from a {class}`~xdem.DEM` (e.g., {func}`xdem.DEM.slope`) or
-through the {class}`~xdem.terrain` module which allows array input.
+through the {class}`~xdem.terrain` module (e.g., {func}`xdem.terrain.slope`) which allows a NumPy array input but
+requires additional metadata.
 
+```{code-cell} ipython3
+:tags: [remove-cell]
+
+# To get a good resolution for displayed figures
+from matplotlib import pyplot
+pyplot.rcParams['figure.dpi'] = 600
+pyplot.rcParams['savefig.dpi'] = 600
+```
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+:mystnb:
+:  code_prompt_show: "Show the opening of example files."
+:  code_prompt_hide: "Hide the opening of example files."
+
+import xdem
+
+# Open a DEM from a filename on disk
+filename_dem = xdem.examples.get_path("longyearbyen_ref_dem")
+dem = xdem.DEM(filename_dem)
+```
+
+```{code-cell} ipython3
+# Slope from DEM method
+slope = dem.slope()
+# Or from terrain module using an array input
+slope = xdem.terrain.slope(dem.data, resolution=dem.res)
+```
+
+If computational performance is key, xDEM can rely on [RichDEM](https://richdem.readthedocs.io/) by specifying
```

Comment:
Maybe add a note that we hope to speed-up calculation (using convoutions)?

Reply:
Maybe you can also mention that if several attributes are needed, for optimization is is also better to call `get_terrain_attribute`? I know it is described later as well, but maybe have all "performance" related points at one location is good.

Reply:
Thank god you commented here, I forgot to remove the RichDEM support since #558.
Rest done!